pkg/schedule/windowed_schedule.go

```diff
@@ -12,35 +31,72 @@ var (
 
 // WindowedSchedule is a schedule that waits on an interval within a specific time window
 type WindowedSchedule struct {
-	Interval  time.Duration
-	StartTime *time.Time
-	StopTime  *time.Time
-	state     ScheduleState
+	Interval   time.Duration
+	StartTime  *time.Time
+	StopTime   *time.Time
+	Count      uint
+	state      ScheduleState
+	stopOnTime *time.Time
 }
 
-// NewWindowedSchedule returns an instance of WindowedSchedule given duration,
-// start and stop time
-func NewWindowedSchedule(i time.Duration, start *time.Time, stop *time.Time) *WindowedSchedule {
+// NewWindowedSchedule returns an instance of WindowedSchedule with given interval, start and stop timestamp
+// and count of expected runs. The value of `count` determines stop time, so specifying it together with `stop`
+// is not allowed and the count will be set to defaults 0 in such cases.
+func NewWindowedSchedule(i time.Duration, start *time.Time, stop *time.Time, count uint) *WindowedSchedule {
+	// if stop and count were both defined, ignore the `count`
+	if count != 0 && stop != nil {
+		count = 0
+		// log about ignoring the `count`
+		logger.WithFields(log.Fields{
+			"_block": "NewWindowedSchedule",
+		}).Warning("Specifying both the window stop and the count is not allowed. The parameter `count` has been ignored.")
```

Comment:
Changed to "The window stop timestamp and the count cannot be specified simultaneously. The parameter `count` has been ignored.". Is it ok?